airflow/auth/managers/base_auth_manager.py

```diff
@@ -278,14 +233,6 @@ def is_authorized_custom_view(self, *, method: ResourceMethod | str, resource_na
         :param user: the user to perform the action on
         """
 
-    @abstractmethod
-    def filter_permitted_menu_items(self, menu_items: list[MenuItem]) -> list[MenuItem]:
```

Comment:
Note that we probably need to create a similar one for the new UI. @pierrejeambrun @bbovenzi

Reply:
For the front-end display of menu items, we do not have a solution settled, we imagined a backend endpoint to retrieve all user permissions, that would be called at the bootstrap of the front-end app. Then the UI can do some logic and display appropriate elements based on that.

The backend will still need to verify such permissions too.

Reply:
I was thinking something slightly different, very similar to the one I am deleting. We should a method in auth manager interface that takes a list of menu items and return the same list but filtered out based on user permissions. A menu item could be just a string which identifies the menu item such "variables", "connections" etc ...

On the backend we could expose an API to retrieve all menu items available to the user and the front-end would use that endpoint to display them